### ISSUE TITLE
The BitAndAssign-like traits should have &Self RHS instead of Self

### ISSUE THREAD

**Issue by flaghacker** [petgraph/fixedbitset#49]
Maybe I'm misunderstanding something, but it looks like it's currently not possible to write code like this:

```
let mut left: FixedBitSet = ...
let right: &FixedBitSet = ...
left &= right;
```
Instead you need to use the equivalent but more verbose 
```
left.intersect_with(right)
```
To me it looks like the `BitAndAssign` and related implementations take the RHS by value for no reason.

I think this can be fixed in a backwards compatible manner by adding new implementations like this:

```
impl <'a> BitAndAssign<Rhs = &Self> for FixedBitSet { ... }
```

Would a pull request for this be accepted?

**Comment by jrraymond** [comment]
`FixedBitSet` implements both the `BitAssign` and the `BitAndAssign` traits. The following code runs:
```
use fixedbitset::FixedBitSet;                   
fn main() {                                       
    let mut left = FixedBitSet::with_capacity(2); 
    left.insert(1);                               
    println!("{:b}", &left);                      
    let mut right = FixedBitSet::with_capacity(5);
    right.insert(1);                              
    right.insert(3);                              
    println!("{:b}", &right);                     
    left &= right;                                
    println!("{:b}", &left);
}
>>>
01
01010
01
```
The `BitAndAssign` implementation calls `intersect_with`

**Comment by flaghacker** [comment]
Yes, it works if the right had side has type `FixedBitSet`, but it doesn't work if it has type `&FixedBitSet`. I suggest adding additional implementations for the reference versions.

**Comment by jrraymond** [comment]
such a PR would be appreciated :)

**Comment by jrraymond** [comment]
This is released in 0.3.1. Thanks for the contribution!
